nix-lib/default.nix

```diff
@@ -109,6 +109,46 @@ let
     in
       builtins.filterSource (file: _: ! (hasAnyPrefix file)) path;
 
+  # `callYarnLock` calls `yarn2nix` to generate a nix representation of
+  # a `yarn.lock` file and directly imports it. It uses `yarn2nix`'s
+  # offline mode, so its resolving capabilities are limited, i. e. git
+  # dependencies are not possible.
+  #
+  # Example usage:
+  #
+  # ```
+  # buildNodeDeps (callYarnLock ./yarn.lock {})
+  # ```
+  callYarnLock = yarnLock: { name ? "yarn.lock.nix" }:
+    pkgs.callPackage (pkgs.runCommand name {
+      # faster to build locally, see also note at linkNodeDeps
+      allowSubstitutes = false;
+      preferLocalBuild = true;
+    } ''
+      ${yarn2nix}/bin/yarn2nix --offline ${yarnLock} > $out
+    '') { };
```

Comment:
I think the last {} has to be the override attrset, no? Aka we shouldn’t just make it a `{}` but pass it through.

```
callYarnLock {
  name = "yarn.lock.nix";
  lock = ./yarn.lock;
} {}
```

Reply:
Nvm, the attrset is just `{ fetchurl, fetchgit }`.